torchao/sparsity/prototype/superblock/blocksparse.py

```diff
@@ -16,9 +17,30 @@ def blocksparse_linear(A: torch.Tensor, crow_indices: torch.Tensor, col_indices:
 
 @torch.library.register_fake("blocksparse::linear")
 def blocksparse_linear_abstract(A: torch.Tensor, crow_indices: torch.Tensor, col_indices: torch.Tensor, values: torch.Tensor, M: int, K:int , bias: torch.Tensor) -> torch.Tensor:
-    new_shape = A.shape[:-1] + (bias.shape[0],)
+    new_shape = A.shape[:-1] + (M,)
     return torch.empty(new_shape, dtype=A.dtype, device=A.device)
 
+@torch.library.custom_op("blocksparse::_int_mm", mutates_args=())
+def blocksparse_int_mm(crow_indices: torch.Tensor,
+                   col_indices: torch.Tensor,
+                   values: torch.Tensor,
+                   M: int,
+                   K: int,
+                   A: torch.Tensor) -> torch.Tensor:
+    assert values.dtype == torch.int8
+    weight_bsr = torch.sparse_bsr_tensor(crow_indices, col_indices, values, size=(M, K))
+                           
+    N = A.shape[-1]
+ 
+    # original_batch_dims_broadcasted = broadcast_batch_dims("_int_bsr_dense_addmm", weight_bsr, A)
+    # input = torch.zeros(M, N, dtype=torch.int32, device=A.device)
+    return bsr_dense_mm(weight_bsr, A).t().contiguous()
```

Comment:
```suggestion
    return bsr_dense_mm(weight_bsr, A).t()
```
That should increase the performance of
```
benchmark.py --model vit_h_14   --batch-size 256   --sparsity-linear 0.8   --sp-linear-tile-size 64 --bsr 64 --sparsity bsr --quantization
```
by ~0.9 img/s.